### DISCUSSION TITLE
Should I stick with the default helix configuration or customize it to resemble Vim?

### DISCUSSION THREAD

**Discussion by logan272** [helix-editor/helix#8585] [Q&A] [answered]
I recently discovered helix and have been absolutely enthralled with it! As someone who previously used Vim/Neovim, I always struggled with maintaining and synchronizing my Vim configurations with the community. I like the simplicity of helix, where everything is seamlessly built-in and just works. What's even better is that the configuration is stored as data, rather than code. I cannot emphasize enough how much I adore this aspect.

While I find the default helix configuration to be excellent, I'm accustomed to Vim motions and now I'm contemplating whether I should stick with the default helix motions and configurations, and retrain my muscle memory. 

How do you think? love to hear you opinions and feedbacks.

Thanks!

**Comment by Omnikar** [comment, marked as answer]
I also switched over to Helix from having Vim muscle memory and my own recommendation would be to just stick with the default motions and take the temporary struggle to get accustomed to the different bindings. You'll get used to it eventually and it'll be worth it when you do.

**Comment by mweihs** [comment]
Since you are already accustomed to the Vim keybindings, I think there is no downside to configuring Helix to be more vim-like.

**Reply by pascalkuthe** [reply]
Helix has a fundamentally different editing model than Vin. If you try to just use helix like it's vim you are nor going to have a good time.

It's not vim, it can't be co figured to behave like vim and we are not interested in supporting vim-like bindings

**Reply by mweihs** [reply]
"More vim-like" doesn't mean "turn Helix into Vim".
e.g. I'm used to '0' and '$' from vim, so I remapped "gh" and "gl". Makes it more vim-like, and I'm having a good time.

**Reply by pascalkuthe** [reply]
Ofcourse small changes to theymap can make sense. I wouldn't really call that making helix work like vim just some slight personal preference for keymaps but this question doesn't sound its about that.

>  I'm accustomed to Vim motions 

Helix motions are just fundamentally different and if you try to force it to act like vim it won't work well.

**Reply by mweihs** [reply]
Maybe I misunderstood [logan272](https://github.com/logan272)'s question.

**Reply by logan272** [reply]
Thank you all for your valuable suggestions. After reviewing the Helix tutor and docs, I have made the decision to stick with the default. My exploration has led me to realize that Helix is fundamentally different from Vim, and attempting to turn it into a more Vim-like editor would not be wise, if not impossible. 

One of the reason I initially considered making Helix more Vim-like is that Vim is everywhere and i probably would still use Vim or Vim motions from time to time, it probably would be annoying to switch Helix and Vim back and forth.
